### PR TITLE
OpenSSL License still applied instead of the Apache 2 license

### LINKED ISSUE #23570: OpenSSL License still applied for some source files
Found in OpenSSL 3.2.1 and master branch.

The following files
* include/openssl/hpke.h
* crypto/hpke/hpke.c
* crypto/ec/asm/ecp_sm2p256-armv8.pl
* crypto/chacha/asm/chacha-loongarch64.pl

still seem to be released under the OpenSSL License instead of the Apache 2 license.

Is this intentional or can the files be released under Apache 2.0 as well?
Right now I assume that formally the OpenSSL license applies for OpenSSL 3 as well.

### PR DESCRIPTION
**Pull request by shahsb**
Found in OpenSSL 3.2.1 and master branch.

The following files

include/openssl/hpke.h
crypto/hpke/hpke.c
crypto/ec/asm/ecp_sm2p256-armv8.pl
crypto/chacha/asm/chacha-loongarch64.pl
still seem to be released under the OpenSSL License instead of the Apache 2 license.

Fixes #23570

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/hpke/hpke.c [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /*
- * Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright Siemens AG 2018-2020
```
Comment (slontis) [inline crypto/hpke/hpke.c:3]: Why is there a Siemens copyright on these files?
The date should also be changed back to 2022-2023..

**Review by slontis — COMMENTED** [review]
Do not change the dates on any of these files.. It indicates when they were created/updated

File: crypto/chacha/asm/chacha-loongarch64.pl [hunk 2]
```diff
@@ -1,8 +1,7 @@
 #! /usr/bin/env perl
-# Author: Min Zhou <zhoumin@loongson.cn>
-# Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+# Copyright 2021-2023 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (levitte) [inline crypto/chacha/asm/chacha-loongarch64.pl:2]: Why the year change?

File: crypto/ec/asm/ecp_sm2p256-armv8.pl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env perl
-# Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+# Copyright 2021-2023 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (levitte) [inline crypto/ec/asm/ecp_sm2p256-armv8.pl:2]: Why the year change?

File: crypto/hpke/hpke.c [hunk 4]
```diff
@@ -1,7 +1,8 @@
 /*
- * Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2023 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (levitte) [inline crypto/hpke/hpke.c:2]: Why the year change?

File: include/openssl/hpke.h [hunk 5]
```diff
@@ -1,7 +1,8 @@
 /*
- * Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2000-2023 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (levitte) [inline include/openssl/hpke.h:2]: Why the year change?

**Review by levitte — CHANGES_REQUESTED** [review]
As far as I can tell from git logs, all the copyright years are correct.  Why change them?

**Comment by levitte** [comment]
A thank you is still due.  Thanks for noticing these license discrepancies...  but, could this PR fix *only* that, please?

File: include/openssl/hpke.h [hunk 6]
```diff
@@ -1,7 +1,8 @@
...
+ * Copyright Siemens AG 2018-2020
  *
- * Licensed under the OpenSSL license (the "License").  You may not use
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
```
Comment (tom-cosgrove-arm) [inline include/openssl/hpke.h:5]: NOTE: this file was introduced in 2022, saying "the OpenSSL license". Could we add something to check that (new) files have the Apache 2.0 licence? (not in this PR - asking generally)

File: crypto/hpke/hpke.c (hunk 1 above)
Comment (shahsb) [reply crypto/hpke/hpke.c]: Done. Addressed.

It was copy paste mistake. Thanks!

**Comment by shahsb** [comment]
> Do not change the dates on any of these files.. It indicates when they were created/updated

Done. Addressed.

It was copy paste mistake. Thanks!

File: crypto/chacha/asm/chacha-loongarch64.pl (hunk 2 above)
Comment (shahsb) [reply crypto/chacha/asm/chacha-loongarch64.pl]: Done. Addressed.

It was copy paste mistake. Thanks!

File: crypto/hpke/hpke.c (hunk 4 above)
Comment (shahsb) [reply crypto/hpke/hpke.c]: Done. Addressed.

It was copy paste mistake. Thanks!

File: include/openssl/hpke.h (hunk 5 above)
Comment (shahsb) [reply include/openssl/hpke.h]: Done. Addressed.

It was copy paste mistake. Thanks!

**Comment by shahsb** [comment]
> A thank you is still due. Thanks for trying to fix these license discrepancies... but, could this PR fix _only_ that, please?

Done.

File: include/openssl/hpke.h (hunk 6 above)
Comment (shahsb) [reply include/openssl/hpke.h:4]: Great Point @tom-cosgrove-arm 

Sure, I have opened a new bug - https://github.com/openssl/openssl/issues/23604 to address the same. Thanks!

**Review by levitte — APPROVED** [review]

**Review by slontis — APPROVED** [review]
Thanks for the PR...

**Comment by t8m** [comment]
Merged to the master and 3.2 branches. Thank you for the contribution.

**Pull request closed without merging**